src/diffpy/labpdfproc/tools.py

```diff
@@ -34,3 +34,36 @@ def set_wavelength(args):
         return WAVELENGTHS[args.anode_type]
     else:
         return WAVELENGTHS["Mo"]
+
+
+def _load_key_value_pair(s):
+    items = s.split("=")
+    key = items[0].strip()
+    if len(items) > 1:
+        value = "=".join(items[1:])
+    return (key, value)
+
+
+def load_user_metadata(args):
+    if args.user_metadata:
+        for item in args.user_metadata:
+            if "=" not in item:
+                raise ValueError("please provide key-value pairs in the format key=value.")
+            key, value = _load_key_value_pair(item)
+            setattr(args, key, value)
+    delattr(args, "user_metadata")
+    return args
+
+
+# Notes, can ignore
```

Comment:
this alll needs to be in a docstring or in the help message (but as I said above, better to give examples), and represented in the tests.  Then delete these comments